jyotisha/panchaanga/temporal/festival/applier/solar.py

```diff
@@ -22,6 +25,23 @@ def assign_all(self):
     self.assign_agni_nakshatra()
 
 
+  def assign_sankranti_punyakaala(self):
+    PUNYA_KAALA = {1: (10, 10), 2: (16, 16), 3: (0, 60), 4: (30, 0), 5: (16, 16), 6: (0, 60),
```

Comment:
So, if I understand correctly, meSha-sankrAnti puNyakAla ranges from transition - 1/6th day to transition + 1/6 day and so on? Strange that it is so seemingly arbitrary. Good to add a comment with such an example (can copy paste from here if what was typed is correct). As always, if you have a reference (even if it is secondary), record it. Anyway, seems too complicated to be turned into a rule in adyatithi.